examples-per-language/php/structural/src/Proxy/Security.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace designPatternsForHumans\structural\Proxy;
+
+
+class Security
+{
+
+    protected $door;
+
+    public function __construct(Door $door)
+    {
+        $this->door = $door;
+    }
+
+    public function open($password)
+    {
+        if ($this->authenticate($password)) {
+            $this->door->open();
+            echo 'Access granted!' . PHP_EOL;
```

Comment:
For me it makes more sense if firstly you echo 'Access granted' then open the door :)